tests/reftests/extrafile.test

```diff
@@ -344,25 +354,27 @@ Bad hash for   - ${BASEDIR}/OPAM/repo/default/packages/good-md5-bad-sha256/good-
 - No changes have been performed
 # Return code 31 #
 ### opam source good-md5-bad-sha256 | '.n"' -> '"'
-[WARNING] Some errors extracting to ${BASEDIR}/good-md5-bad-sha256.1: Failure("Bad hash for   - ${BASEDIR}/OPAM/repo/default/packages/good-md5-bad-sha256/good-md5-bad-sha256.1/files/p.patch")
+[WARNING] Some errors extracting to ${BASEDIR}/good-md5-bad-sha256.1:
+            good-md5-bad-sha256.1: Bad hash for extra-file
+            - p.patch
+
 
 ### test -f good-md5-bad-sha256.1/p.patch
 ### rm -r good-md5-bad-sha256.1
 ### opam source good-md5-bad-sha256 --require-checksums | '.n"' -> '"'
-[WARNING] Some errors extracting to ${BASEDIR}/good-md5-bad-sha256.1: Failure("Bad hash for   - ${BASEDIR}/OPAM/repo/default/packages/good-md5-bad-sha256/good-md5-bad-sha256.1/files/p.patch")
+[WARNING] Some errors extracting to ${BASEDIR}/good-md5-bad-sha256.1:
+            good-md5-bad-sha256.1: Bad hash for extra-file
+            - p.patch
+
 
 ### test -f good-md5-bad-sha256.1/p.patch
 ### opam clean --download-cache
 Clearing cache of downloaded files
 ### :I:5: no checksum
 ### opam lint --package no-checksum
-<default>/no-checksum.1: Errors.
-             error  3: File format error in 'extra-files' at line 11, column 2: expected [file checksum]
-# Return code 1 #
+<default>/no-checksum.1: Passed.
```

Comment:
This change is due to the fact that now we take the opam file from repo state instead of the plain file from opam internal repo copy. Stored opam file in repo state removes the mismatching extra files, so there is no longer the error (extra-files field removed).